src/System.Private.ServiceModel/src/System/IdentityModel/Tokens/KeyInfoSerializer.cs

```diff
@@ -0,0 +1,372 @@
+            {
+                if ((entry is WSSecurityXXX2005) || (entry is WSSecurityJan2004))
+                {
+                    wsSecuritySerializerFound = true;
+                    break;
+                }
+            }
+
+            if (!wsSecuritySerializerFound)
+            {
+                _serializerEntries.Add(new WSSecurityXXX2005((WSSecurityTokenSerializer) innerSecurityTokenSerializer));
+            }
+
+            _tokenEntries = new List<TokenEntry>();
+            _keyIdentifierEntries = new List<SecurityTokenSerializer.KeyIdentifierEntry>();
+            _keyIdentifierClauseEntries = new List<SecurityTokenSerializer.KeyIdentifierClauseEntry>();
+
+            for (int i = 0; i < _serializerEntries.Count; ++i)
+            {
+                SecurityTokenSerializer.SerializerEntries serializerEntry = _serializerEntries[i];
+                serializerEntry.PopulateTokenEntries(_tokenEntries);
+                serializerEntry.PopulateKeyIdentifierEntries(_keyIdentifierEntries);
+                serializerEntry.PopulateKeyIdentifierClauseEntries(_keyIdentifierClauseEntries);
+            }
+        }
+
+        public DictionaryManager DictionaryManager
+        {
+            get { return _dictionaryManager; }
+        }
+
+        /// <summary>
+        /// Gets or sets a value indicating if BSP required attributes should be written out.
+        /// </summary>
+        public bool EmitBspRequiredAttributes
+        {
+            get
+            {
+                return _emitBspRequiredAttributes;
+            }
+        }
+
+        public SecurityTokenSerializer InnerSecurityTokenSerializer
+        {
+            get
+            {
+                return _innerSecurityTokenSerializer == null ? this : _innerSecurityTokenSerializer;
+            }
+           set
+            {
+                _innerSecurityTokenSerializer = value;
+            }
+        }
+
+        protected override bool CanReadTokenCore(XmlReader reader)
+        {
+            return false;
+        }
+
+        protected override SecurityToken ReadTokenCore(XmlReader reader, SecurityTokenResolver tokenResolver)
+        {
+            XmlDictionaryReader localReader = XmlDictionaryReader.CreateDictionaryReader( reader ); 
+            throw ServiceModel.DiagnosticUtility.ExceptionUtility.ThrowHelperError( new XmlException( SR.Format( SR.CannotReadToken, reader.LocalName, reader.NamespaceURI, localReader.GetAttribute( XD.SecurityJan2004Dictionary.ValueType, null ) ) ) );
+        }
+
+        protected override bool CanWriteTokenCore(SecurityToken token)
+        {
+            return false;
+        }
+
+        protected override void WriteTokenCore(XmlWriter writer, SecurityToken token)
+        {
+            throw ServiceModel.DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.StandardsManagerCannotWriteObject, token.GetType())));
```

Comment:
>ServiceModel [](start = 18, length = 12)

NIT: Consider aliasing `using DiagnosticUtility = System.ServiceModel.DiagnosticUtility` to avoid this weirdness of having to prefix DiagnosticUtility... not a usage pattern we use typically isall